hummingbot/connector/exchange/binance/binance_api_order_book_data_source.py

```diff
@@ -44,97 +45,113 @@ def logger(cls) -> HummingbotLogger:
                            domain: str = "com") -> Dict[str, Any]:
-        params: Dict = {"limit": str(limit), "symbol": convert_to_exchange_trading_pair(trading_pair)} if limit != 0 \
-            else {"symbol": convert_to_exchange_trading_pair(trading_pair)}
-        url = SNAPSHOT_REST_URL.format(domain)
-        async with client.get(url, params=params) as response:
-            response: aiohttp.ClientResponse = response
-            if response.status != 200:
-                raise IOError(f"Error fetching market snapshot for {trading_pair}. "
-                              f"HTTP status is {response.status}.")
-            data: Dict[str, Any] = await response.json()
-
-            # Need to add the symbol into the snapshot message for the Kafka message queue.
-            # Because otherwise, there'd be no way for the receiver to know which market the
-            # snapshot belongs to.
-
-            return data
+        throttler = BinanceAPIOrderBookDataSource._get_throttler_instance()
+        params: Dict = {"limit": str(limit), "symbol": binance_utils.convert_to_exchange_trading_pair(trading_pair)} if limit != 0 \
+            else {"symbol": binance_utils.convert_to_exchange_trading_pair(trading_pair)}
+        async with aiohttp.ClientSession() as client:
+            async with throttler.execute_task(limit_id=CONSTANTS.SNAPSHOT_PATH_URL):
+                url = binance_utils.public_rest_url(path_url=CONSTANTS.SNAPSHOT_PATH_URL, domain=domain)
+                async with client.get(url, params=params) as response:
+                    response: aiohttp.ClientResponse = response
+                    if response.status != 200:
+                        raise IOError(f"Error fetching market snapshot for {trading_pair}. "
+                                      f"Response: {response}.")
+                    data: Dict[str, Any] = await response.json()
+
+                    # Need to add the symbol into the snapshot message for the Kafka message queue.
+                    # Because otherwise, there'd be no way for the receiver to know which market the
+                    # snapshot belongs to.
+
+                    return data
 
     async def get_new_order_book(self, trading_pair: str) -> OrderBook:
-        async with aiohttp.ClientSession() as client:
-            snapshot: Dict[str, Any] = await self.get_snapshot(client, trading_pair, 1000, self._domain)
-            snapshot_timestamp: float = time.time()
-            snapshot_msg: OrderBookMessage = BinanceOrderBook.snapshot_message_from_exchange(
-                snapshot,
-                snapshot_timestamp,
-                metadata={"trading_pair": trading_pair}
-            )
-            order_book = self.order_book_create_function()
-            order_book.apply_snapshot(snapshot_msg.bids, snapshot_msg.asks, snapshot_msg.update_id)
-            return order_book
+        snapshot: Dict[str, Any] = await self.get_snapshot(trading_pair, 1000, self._domain)
```

Comment:
In this case, we're able to pass the throttler of the instance to `get_snapshot`